File: UPTMDigital.API/Models/Estudiante.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UPTMDigital.API.Models
{
    [Table("Estudiante")]
    public class Estudiante
    {
        [Key]
        public int IdEstudiante { get; set; }
        public string Cedula { get; set; } = null!;
        public string Nombres { get; set; } = null!;
        public string Apellidos { get; set; } = null!;
        public string? CorreoInstitucional { get; set; }
        public string? CodAlumno { get; set; }
        public string? Telefono { get; set; }
        public string? Direccion { get; set; }

        // --- FK Normalizadas ---
        public int? UsuarioId { get; set; }
        [ForeignKey("UsuarioId")]
        public Usuario? Usuario { get; set; }

        public int? CarreraId { get; set; }
        [ForeignKey("CarreraId")]
        public Carrera? Carrera { get; set; }

        // Campo calculado/legacy para compatibilidad
        [NotMapped]
        public string CarreraNombre => Carrera?.Nombre ?? "";

        public DateTime? FechaRegistro { get; set; }

        /// <summary>Estado del arancel: "Solvente" o "Pendiente". Controla acceso al carnet y constancias.</summary>
        public string EstadoArancel { get; set; } = "Pendiente";

        public bool Activo { get; set; } = true;
    }
}

File: uptmdigital_app/lib/models/estudiante.dart
class Estudiante {
  final int idEstudiante;
  final String cedula;
  final String nombres;
  final String apellidos;
  final String correoInstitucional;
  final int? usuarioId;
  final int? carreraId;
  final String carreraNombre;
  final String? telefono;
  final String? profileImageUrl;

  Estudiante({
    required this.idEstudiante,
    required this.cedula,
    required this.nombres,
    required this.apellidos,
    required this.correoInstitucional,
    this.usuarioId,
    this.carreraId,
    this.carreraNombre = '',
    this.telefono,
    this.profileImageUrl,
  });

  factory Estudiante.fromJson(Map<String, dynamic> json) {
    // carrera can be a nested object { nombre: "..." } or null
    String carrera = '';
    if (json['carrera'] is Map) {
      carrera = json['carrera']['nombre'] ?? '';
    } else if (json['carreraNombre'] is String) {
      carrera = json['carreraNombre'];
    }

    return Estudiante(
      idEstudiante: json['idEstudiante'] ?? 0,
      cedula: json['cedula'] ?? '',
      nombres: json['nombres'] ?? '',
      apellidos: json['apellidos'] ?? '',
      correoInstitucional: json['correoInstitucional'] ?? '',
      usuarioId: json['usuarioId'],
      carreraId: json['carreraId'],
      carreraNombre: carrera,
      telefono: json['telefono'],
      profileImageUrl: json['profileImageUrl'],
    );
  }
}
